app/models/host/managed.rb

```diff
@@ -916,6 +908,16 @@ def apply_compute_profile(modification)
     modification.run(self, compute_resource.try(:compute_profile_for, compute_profile_id))
   end
 
+  def puppet_classes
```

Comment:
This is very similar to `#puppetclasses` from the association, but it returns very different structures. I'd suggest naming it `enc_puppetclasses` or similar.